Drivers/c/LidarExample/main.c

```diff
@@ -0,0 +1,175 @@
+#include "FreeRTOS.h"
+#include "task.h"
+
+#include <Hal/Hal.h>
+#include <Init/Init.h>
+
+#include <RegisterHwas/RegisterHwas.h>
+#include <UartHwas/UartHwas.h>
+#include <UartHwas/UartHwasRegisters.h>
+
+#include <stdio.h>
+#include <string.h>
+
+#define LIDAR_RX_BUFFER_LEN 100
+uint8_t lidarRxBuffer[LIDAR_RX_BUFFER_LEN] = {""};
+
+ByteFifo rxFifo;
+UartHwas uart;
+
+#define PRV_INTERRUPT_STACK_SIZE 400
+__attribute__((section(".sdramMemorySection"))) static StackType_t
+    prvTaskStackBuffer[PRV_INTERRUPT_STACK_SIZE];
+__attribute__((
+    section(".sdramMemorySection"))) static StaticTask_t prvTaskBuffer;
+
+#define TF_LUNA_DATA_FRAME_HEADER1 0x59
+#define TF_LUNA_DATA_FRAME_HEADER2 0x59
+#define TF_LUNA_DATA_FRAME_LENGTHT 9
+
+size_t txIndex = 0;
+size_t txLength = 0;
+uint8_t txBuffer[20];
+
+static void prvTask(void *pvParameters);
+
+void lidarRxCallback(uint8_t readByte) {
+  if (!ByteFifo_push(&rxFifo, readByte)) {
+    Hal_console_usart_write("Fifo Full\n", 10);
+  }
+}
+
+void lidarTxCallback(void) {
+  ++txIndex;
+  if (txIndex < txLength) {
+    UartHwas_sendByteAsync(&uart, lidarTxCallback, txBuffer[txIndex]);
+  } else {
+    Register_set_bits(uart.uartAddress + UART_HWAS_IDR_OFFSET,
+                      UART_HWAS_IxR_TXRDY_MASK | UART_HWAS_IxR_TXEMPTY_MASK);
+  }
+}
+
+extern void
+hwas_RI_InterruptSubscription_Interrupt_Ri(const asn1SccInterrupt_Type *irq);
+void hwas_RI_InterruptSubscription_Interrupt_Ri(
+    const asn1SccInterrupt_Type *irq) {
+
+  switch (irq->interrupt) {
+  case Nvic_Irq_Uart0:
+  case Nvic_Irq_Uart1:
+  case Nvic_Irq_Uart2:
+  case Nvic_Irq_Uart3:
+  case Nvic_Irq_Uart4:
+    UartHwas_handleInterrupt(&uart);
+    hwas_PI_InterruptManagement_EnableInterrupt_Pi(&irq->interrupt);
+  }
+}
+
+int main() {
+  Init_setup_hardware();
+  Hal_console_usart_init();
+
+  hwas_startup();
+  xTaskCreateStatic(prvTask, "Task1", 400, NULL, 1, prvTaskStackBuffer,
+                    &prvTaskBuffer);
+
+  vTaskStartScheduler();
+  for (;;) {
+  }
+}
+
+static void parseTfLunaData() {
+  uint8_t rxByte;
+  uint8_t checksum;
```

Comment:
I think that reliance on unsigned int 8 overflow for modulo is not a great idea. Could we change this to e.g. uint32_t and then just do %?